app/src/main/java/org/oppia/app/administratorcontrols/administratorcontrolsitemviewmodel/AdministratorControlsDownloadPermissionsViewModel.kt

```diff
@@ -1,4 +1,38 @@
 package org.oppia.app.administratorcontrols.administratorcontrolsitemviewmodel
 
+import androidx.databinding.ObservableField
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import org.oppia.app.model.DeviceSettings
+import org.oppia.app.model.ProfileId
+import org.oppia.domain.profile.ProfileManagementController
+import org.oppia.util.logging.Logger
+
 /** [ViewModel] for the recycler view in [AdministratorControlsFragment]. */
-class AdministratorControlsDownloadPermissionsViewModel : AdministratorControlsItemViewModel()
+class AdministratorControlsDownloadPermissionsViewModel(
+  private val fragment: Fragment,
+  private val logger: Logger,
+  private val profileManagementController: ProfileManagementController,
+  private val userProfileId: ProfileId,
+  private val deviceSettings: DeviceSettings
+) : AdministratorControlsItemViewModel() {
+
+  var isTopicWifiUpdatePermission = ObservableField<Boolean>(deviceSettings.allowDownloadAndUpdateOnlyOnWifi)
+  var isTopicAutoUpdatePermission = ObservableField<Boolean>(deviceSettings.automaticallyUpdateTopics)
+
+  fun onTopicWifiUpdatePermissionChanged(checked: Boolean) {
+    profileManagementController.updateWifiPermissionDeviceSettings(userProfileId, checked).observe(fragment, Observer {
+      if (it.isFailure()) {
+        logger.e("AdministratorControlsDownloadPermissionsViewModel", "Failed to update topic update on wifi permission", it.getErrorOrNull()!!)
+      }
+    })
+  }
+
+  fun onTopicAutoUpdatePermissionChanged(checked: Boolean) {
+    profileManagementController.updateTopicAutomaticallyPermissionDeviceSettings(userProfileId, checked).observe(fragment, Observer {
+      if (it.isFailure()) {
+        logger.e("AdministratorControlsDownloadPermissionsViewModel", "Failed to update topic auto update permission", it.getErrorOrNull()!!)
```

Comment:
Keep fragment name in place of TAG here and above.

Reply:
Done

Reply:
Done